### ISSUE TITLE
UIA Checkboxes unrecognized by NVDA

### ISSUE THREAD

**Issue by yaakovschectman** [flutter/flutter#119350]
Currently in our UIA migration, checkbox widgets are not treated by NVDA as checkboxes, even when they have the checkbox role and provide the toggle pattern. I do not currently know why that is. I have filed [an issue](https://github.com/nvaccess/nvda/issues/14584) on the NVDA repository in the hopes of uncovering more information. Windows Narrator responds as expected to checkboxes.

There are three cases that I am/have been invetigating:

- When hovering the cursor over a checkbox, nothing is read indicating it is a checkbox.
  - This is apparently intended behavior with NVDA
- When changing the state of a focused checkbox, nothing is read to indicate the change of state.
  - This can be addressed by modifying `AXPlatformNodeWin::MojoEventToUIAProperty`
- When the user pressed NVDA+UP with a checkbox focused, the checkbox info is not read aloud.
  - This should be the case, as NVDA+UP is only meant to read textual information. NVDA+TAB is a separate keyboard command for reading the current focus, and it does in fact read the checkbox state.

**Comment by yaakovschectman** [comment]
As cases 1 and 3 above were determined to be non-issues, and case 2 is now resolved by the above PR, I will mark this resolved.
